### ISSUE TITLE
access invalid pointer 0x000000000001

### ISSUE THREAD

**Issue by charlesxsh** [facebookexperimental/starlark-rust#25]
For given input file in the zip file, the code compiled with AddressSanitizer reported invalid address access error during runtime.
[in.zip](https://github.com/facebookexperimental/starlark-rust/files/6818410/in.zip)

Code
```rust
use starlark::eval::Evaluator;
use starlark::environment::{Module, Globals};
use starlark::values::Value;
use starlark::syntax::{AstModule, Dialect};


fn main() {
    // change filepath to the input file
    let data = std::fs::read(filepath).unwrap();
        let _  = match std::str::from_utf8(&data) {
            Ok(d) => {
                if let Ok(ast) = AstModule::parse("hello_world.star", d.to_owned(), &Dialect::Standard) {
                   
                    let globals: Globals = Globals::standard();

                    let module: Module = Module::new();

                    let mut eval: Evaluator = Evaluator::new(&module, &globals);

                    eval.eval_module(ast);  
                }

                
            },
            Err(..) => return,
        };
       
}
```

Output
```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==21690==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000001 (pc 0x55ac52e1190a bp 0x7ffffab48110 sp 0x7ffffab47e80 T0)
==21690==The signal is caused by a READ memory access.
==21690==Hint: address points to the zero page.
    #0 0x55ac52e1190a in starlark::values::fast_string::skip_at_most_1byte::f::h0e1da77a9859a9a1 /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/values/fast_string.rs:55:34
    #1 0x55ac52e11255 in starlark::values::fast_string::skip_at_most_1byte::h40f70214e41aab2e /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/values/fast_string.rs:84:14
    #2 0x55ac52e11f94 in starlark::values::fast_string::at::h317771503d053ab2 /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/values/fast_string.rs:89:13
    #3 0x55ac53003f9e in starlark::values::types::string::_$LT$impl$u20$starlark..values..traits..StarlarkValue$u20$for$u20$alloc..boxed..Box$LT$str$GT$$GT$::at::hd2466a6c38857b75 /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/values/types/string.rs:180:27
    #4 0x55ac530468b2 in starlark::values::_$LT$impl$u20$starlark..values..layout..value..Value$GT$::at::h86d1c57ef797f103 /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/values/mod.rs:454:9
    #5 0x55ac52cc6d9b in starlark::eval::expr::_$LT$impl$u20$starlark..eval..Compiler$GT$::expr::_$u7b$$u7b$closure$u7d$$u7d$::h97225333fb9eecfe /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/eval/expr.rs:434:25
    #6 0x55ac52a38b3c in _$LT$alloc..boxed..Box$LT$F$GT$$u20$as$u20$core..ops..function..Fn$LT$A$GT$$GT$::call::hfb7414494fc44b22 /home/xsh/code/rust-compiler/src/liballoc/boxed.rs:1095:9
    #7 0x55ac52d13f27 in starlark::eval::stmt::_$LT$impl$u20$starlark..eval..Compiler$GT$::stmt::_$u7b$$u7b$closure$u7d$$u7d$::h823a805583856ead /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/eval/stmt.rs:423:21
    #8 0x55ac52a38b3c in _$LT$alloc..boxed..Box$LT$F$GT$$u20$as$u20$core..ops..function..Fn$LT$A$GT$$GT$::call::hfb7414494fc44b22 /home/xsh/code/rust-compiler/src/liballoc/boxed.rs:1095:9
    #9 0x55ac52d407e1 in starlark::eval::_$LT$impl$u20$starlark..eval..context..Evaluator$GT$::eval_module::h1885c10571a8c91a /home/xsh/.cargo/registry/src/github.com-1ecc6299db9ec823/starlark-0.4.0/src/eval/mod.rs:168:19
    #10 0x55ac53306ddb in starlarkfuzzvrf::main::hfa9f9430d0214a86 /home/xsh/code/rust-asan/fuzz/starlark-vrf.rs:30:21
    #11 0x55ac53306986 in std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::hf7e2a6fbfb396fba /home/xsh/code/rust-compiler/src/libstd/rt.rs:67:34
    #12 0x55ac53863612 in std::rt::lang_start_internal::h32d9533fcd93e198 (/home/xsh/code/rust-asan/fuzz/target/shihao/x86_64-unknown-linux-gnu/debug/starlarkfuzzvrf+0x1690612)
    #13 0x55ac53306ac5 in std::rt::lang_start::hddcbf92bd8879287 /home/xsh/code/rust-compiler/src/libstd/rt.rs:67:5
    #14 0x55ac53306af9 in main (/home/xsh/code/rust-asan/fuzz/target/shihao/x86_64-unknown-linux-gnu/debug/starlarkfuzzvrf+0x1133af9)
    #15 0x7fd6c03e9bf6 in __libc_start_main /build/glibc-S9d2JN/glibc-2.27/csu/../csu/libc-start.c:310
    #16 0x55ac523a06f9 in _start (/home/xsh/code/rust-asan/fuzz/target/shihao/x86_64-unknown-linux-gnu/debug/starlarkfuzzvrf+0x1cd6f9)
```

**Comment by ndmitchell** [comment]
To confirm, the input file reads: `""[2]` - that makes sense, I can see why this code might trigger a bug. Thanks for the excellent report, I'll investigate now.

**Comment by ndmitchell** [comment]
Thanks for the report, investigating now.

**Comment by ndmitchell** [comment]
I'm pretty sure this is the same route cause and fix as #24, even though the final error hit is different, so merging with that.
